cpp/src/cylon/net/ucc/ucc_operations.cpp

```diff
@@ -259,17 +262,25 @@ Status UccTableGatherImpl::GatherBufferSizes(const int32_t *send_data, int32_t n
   ucc_coll_req_h req;
 
   args.mask = 0;
-  args.coll_type = UCC_COLL_TYPE_GATHER;
+  args.coll_type = UCC_COLL_TYPE_ALLGATHER;
```

Comment:
put a `TODO` and refer to the github issue. 
Add a comment explaining why we are using allgather